src/adapter.js

```diff
@@ -40,20 +39,18 @@
         if(!karma.config.jspm.useBundles){
             System.bundles = [];
         }
-        
-        // Load everything specified in loadFiles
-        for (var i = 0; i < karma.config.jspm.expandedFiles.length; i++) {
-            var modulePath = karma.config.jspm.expandedFiles[i];
-            var promise = System['import'](extractModuleName(modulePath))
-                ['catch'](function(e) {
-                    throw e;
-                });
-            promises.push(promise);
-        }
 
-        // Promise comes from the systemjs polyfills
-        Promise.all(promises).then(function() {
+        // Load everything specified in loadFiles in the specified order
+        var promiseChain = karma.config.jspm.expandedFiles.reduce(function (prev, next) {
```

Comment:
it would be nice to maintain that backwards compatibility (though I'm not sure how many users still require it).  could just include the polyfill:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#Polyfill